shared/actions/git/index.js

```diff
@@ -147,6 +147,19 @@ const _onTabChange = (action: RouteTreeConstants.SwitchTo) => {
   return null
 }
 
+function* _handleIncomingGregor(action: Constants.HandleIncomingGregor): SagaGenerator<any, any> {
+  const msgs = action.payload.messages.map(msg => JSON.parse(msg.body))
+  for (let body of msgs) {
+    switch (body.action) {
+      case 'delete': // fallthrough
+      case 'create': // fallthrough
+      case 'update': // Reload git info on actions
+        yield put(Creators.loadGit())
```

Comment:
Is there a reason we'd want to do this more than once instead of returning?

Reply:
agree, you could also do
```
const needsLoad = ['delete', 'create', 'update'].includes(body.action)
```
instead of this switch statement

Reply:
:+1:

Reply:
As for the `return`ing: I don't want to discount that something other than a load would be needed for other messages. In this case that's the only possibility though, so a `return` works out